server/neptune/temporalworker/server.go

```diff
@@ -185,20 +196,32 @@ func (s Server) Start() error {
 	}()
 
 	// Start job output handler listener
-	// [WENGINES-4746] TODO: Clean up resources and exit gracefully on SIGTERM
 	go func() {
+		// Exits when the job output chan is closed which happens only after the temporal worker
+		// has gracefully shut down
 		s.JobStreamHandler.Handle()
 	}()
 
 	<-stop
 	wg.Wait()
 
+	// Close job stream when temporalworker exits to allow gracefully shutting down the stream handler
+	s.Logger.Info("Closing job output channel")
+	close(s.JobStream)
```

Comment:
What happens if handle has not exited before cleanup is called?

You should be using a wait group with the handle go routine.  and reading your other response, it probably does make sense to keep the Closing of the global channel in the worker go routine given that dependence.

Reply:
Added a wait group to the handle goroutine!